src/api/posts.ts

```diff
@@ -0,0 +1,46 @@
+import { UserComment } from '../types/typed';
```

Comment:
rename this file to api.ts, you have here methods not only got posts, but for users and comments too